Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/RestShapeSpringsForceField.inl

```diff
@@ -470,18 +472,18 @@ void RestShapeSpringsForceField<DataTypes>::addKToMatrix(const MechanicalParams*
 
         // translation
         const auto vt = -kFact * (index < k.size() ? k[index] : k[0]);
-        for (int i = 0; i < 3; i++)
+        for (int i = 0; i < space_size; i++)
         {
-            mat->add(offset + N * curIndex + i, offset + N * curIndex + i, vt);
+            mat->add(offset + total_size * curIndex + i, offset + total_size * curIndex + i, vt);
         }
 
         // rotation (if applicable)
-        const auto vr = -kFact * (index < k_a.size() ? k_a[index] : k_a[0]);
         if constexpr (isRigidType<DataTypes>())
         {
-            for (int i = 3; i < 6; i++)
+            const auto vr = -kFact * (index < k_a.size() ? k_a[index] : k_a[0]);
```

Comment:
Not a real suggestion, just for fun:

For a branchless computation:
```cpp
const auto vr = -kFact * k_a[(index < k_a.size()) * index];

Reply:
nice trick 👍
lets do the same (addforce, adddforce)

Reply:
@hugtalbot has been traumatized by casts 😂

Reply:
In this case, the cast (`bool` to `decltype(index)`) is implicit and the conversion is not narrowing. You won't get a compilation warning.
But you are also right: the cast is implicit, therefore it may be more difficult to read. And it's often the case with branchless programming. That is why I specified "Not a real suggestion, just for fun"